### PR TITLE
DMT-23 Create build related CI pipeline

### PR DESCRIPTION
**Pull request by eemilj**
## Proposed changes
- Set up the CI pipeline to install and run the node.js build to ensure tha the build succeeds on push and pull requests 

## Additional information
- Using github actions workflow to run our pipeline, tests to be added later

### CODE REVIEW

**Review by clementinejensen — APPROVED** [review]
Looks great! Seems like everything is in order. 👍

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        node-version: [14.x, 16.x]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+      - run: npm install
```
Comment (krasen86) [inline .github/workflows/node.js.yml:23]: Shouldn't be running "npm start" here? So we actually run the package.json file?

Comment (eemilj) [reply .github/workflows/node.js.yml:23]: We had that initially, but that actually starts running the server which will end up running forever, so we decided to only install in this commit.

Comment (krasen86) [reply .github/workflows/node.js.yml:23]: Strange the npm server should exit automatically in the current version, at least that is what's happening when it's run locally. But yeah it makes sense to just install it if the pipeline runs forever :)

**Review by krasen86 — APPROVED** [review]

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (chrytsa) [reply .github/workflows/node.js.yml:23]: The initial version of `package.json` forced Node.js to exit after the corresponding printing to the console. However, the current version on `sprint-1` ([ref](https://github.com/Donut-Mod-Team/donut-mod/blob/6d1042b5deb97b3a33b0ed6677c3679bd5c0a014/package.json#L7)) specified a different execution process, hence it was needed either to be changed in the `package.json` file that in the future will be extended, or at the `node.js.yml`.

Comment (krasen86) [reply .github/workflows/node.js.yml:23]: That's a good point maybe we should have the pipeline run on pull requests as well in order to catch such differences between branches that can catch such errors?

Comment (chrytsa) [reply .github/workflows/node.js.yml:23]: There wasn't any difference or an error. Either we would have to fix a merge conflict while trying to merge to `sprint-1` if we would have decided to keep the older version of `package.json`, or approach the possible solution differently, as we did.

The pipeline runs or pull requests as well. In [that commit](https://github.com/Donut-Mod-Team/donut-mod/pull/7/commits/40260be95126029e2d9cc61962eefa5fe796ac73), you can see that we replaced the particular line that you probably referring to, as it actually triggered double the amount of checks that were redundant. Unless, you mean something different. :stuck_out_tongue:

**Pull request merged**